pkg/terraform/gather/ovirt/ip.go

```diff
@@ -69,52 +76,54 @@ func findVirtualMachineIP(c *ovirtsdk4.Connection, moRefValue string) (string, e
 				ipres, hasAddress := ip.Address()
 				if hasAddress {
 					if checkPortIsOpen(ipres, bootstrapSSHPort) {
-						logrus.Debugf("ovirt vm id: %s , found usable bootstrap IP %s", moRefValue, ipres)
+						logrus.Debugf("ovirt vm id: %s , found usable  IP Address: %s", moRefValue, ipres)
 						return ipres, nil
 					}
 				}
 			}
 		}
 	}
-	return "", fmt.Errorf("could not find usable bootstrap IP address for vm id: %s", moRefValue)
+	return "", fmt.Errorf("could not find usable IP address for vm id: %s", moRefValue)
 }
 
-// BootstrapIP returns the ip address for bootstrap host.
-// still unsupported, because qemu-ga is not available - see https://bugzilla.redhat.com/show_bug.cgi?id=1764804
-func BootstrapIP(tfs *terraform.State) (string, error) {
+func lookupVMResources(tfs *terraform.State,moduleName string,name string) ([]string,error){
 
-	client, err := ovirt.NewConnection()
+	br, err := terraform.LookupResource(tfs, moduleName, "ovirt_vm", name)
 	if err != nil {
-		return "", fmt.Errorf("failed to initialize connection to ovirt-engine's %s", err)
-	}
-	defer client.Close()
-
-	br, err := terraform.LookupResource(tfs, "module.bootstrap", "ovirt_vm", "bootstrap")
-	if err != nil {
-		return "", errors.Wrap(err, "failed to lookup bootstrap")
+		return []string{""}, errors.Wrapf(err, "failed to lookup %s VM",name)
 	}
 
 	if len(br.Instances) == 0 {
-		return "", errors.New("no bootstrap instance found")
+		return []string{""}, errors.New(fmt.Sprintf("no %s instance found",name))
```

Comment:
Returning an empty IP seems like a workaround for something. It seems that this should not be handled here. If yes, we should add comments explaining what this does and why it is ok to return an empty string IP.

Reply:
okay , in case of error we can return nil instead

Reply:
done